### PR TITLE
Add a caution for named wait handles about them being operable by multiple users by default

### PR DESCRIPTION
**Pull request by kouvel**
- There is a potential for issues around using named mutexes in multi-user environments where some users may be untrusted, where an untrusted user may do undesirable things, this should be documented.
- Added additional info for named mutexes on Unix-like OSes regarding other users being able to interfere with them.

### CODE REVIEW

File: xml/System.Threading/EventWaitHandleAcl.xml [hunk 1]
```diff
@@ -78,6 +78,9 @@
...
 
  If a `name` is provided and a synchronization object of the requested type already exists in the namespace, the existing synchronization object is opened. If a synchronization object of a different type already exists in the namespace, a `WaitHandleCannotBeOpenedException` is thrown. Otherwise, a new synchronization object is created.
+  
+> [!CAUTION]
+>  By default, a named event is not restricted to the user that created it. Other users may be able to open and use the event, including interfering with the event such as setting or resetting it inappropriately. An option is to pass in an <xref:System.Security.AccessControl.EventWaitHandleSecurity> when creating the named event to restrict access to specific users. Avoid using named events without access restrictions on systems that may have untrusted users running code.
```
Comment (mangod9) [inline xml/System.Threading/EventWaitHandleAcl.xml:83]: is this across all OSes ?

Comment (kouvel) [reply xml/System.Threading/EventWaitHandleAcl.xml:83]: On Unixes we only have support for named mutexes, so the docs here for events are Windows-specific.

File: xml/System.Threading/EventWaitHandle.xml [hunk 2]
```diff
@@ -75,6 +75,9 @@ The <xref:System.Threading.EventWaitHandle> class allows threads to communicate
   
  For more information, see the [Thread interaction, or signaling](/dotnet/standard/threading/overview-of-synchronization-primitives#thread-interaction-or-signaling) section of the [Overview of synchronization primitives](/dotnet/standard/threading/overview-of-synchronization-primitives) article.
   
+> [!CAUTION]
+>  By default, a named event is not restricted to the user that created it. Other users may be able to open and use the event, including interfering with the event such as setting or resetting it inappropriately. An option is to use a constructor overload or <xref:System.Threading.EventWaitHandleAcl> and pass in an <xref:System.Security.AccessControl.EventWaitHandleSecurity> when creating the named event to restrict access to specific users. Avoid using named events without access restrictions on systems that may have untrusted users running code.
```
Comment (gewarren) [inline xml/System.Threading/EventWaitHandle.xml:79]:
```suggestion
> By default, a named event is not restricted to the user that created it. Other users may be able to open and use the event, including interfering with the event by setting or resetting it inappropriately. To restrict access to specific users, you can use a constructor overload or <xref:System.Threading.EventWaitHandleAcl> and pass in an <xref:System.Security.AccessControl.EventWaitHandleSecurity> when creating the named event. Avoid using named events without access restrictions on systems that might have untrusted users running code.
```

**Review by gewarren — APPROVED** [review]
Can you make the suggested changes in all the other similar places too?

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by kouvel** [comment]
Missed one thing, updaing

**Pull request merged**